upup/pkg/fi/cloudup/awstasks/iamrolepolicy.go

```diff
@@ -192,12 +279,23 @@ func (e *IAMRolePolicy) policyDocumentString() (string, error) {
 }
 
 type terraformIAMRolePolicy struct {
-	Name           *string            `json:"name"`
+	Name           *string            `json:"name,omitempty"`
 	Role           *terraform.Literal `json:"role"`
-	PolicyDocument *terraform.Literal `json:"policy"`
+	PolicyDocument *terraform.Literal `json:"policy,omitempty"`
+	PolicyArn      *string            `json:"policy_arn,omitempty"`
 }
 
 func (_ *IAMRolePolicy) RenderTerraform(t *terraform.TerraformTarget, a, e, changes *IAMRolePolicy) error {
+	if e.ExternalPolicies != nil && len(*e.ExternalPolicies) > 0 {
+		for _, policy := range *e.ExternalPolicies {
+			tf := &terraformIAMRolePolicy{
+				Role:      e.Role.TerraformLink(),
+				PolicyArn: s(policy),
+			}
+			return t.RenderResource("aws_iam_role_policy_attachment", *e.Name, tf)
```

Comment:
This might be related to the resource not showing up in kubernetes.tf, but I think we dont want to return here. We want to render both this `aws_iam_role_policy_attachment` and the `aws_iam_role_policy` resource below. I think we may also need to make the resource name (currently `*e.Name` unique so that multiple `aws_iam_role_policy_attachment` resources dont conflict. Perhaps the name can include a normalized version of the policy arn?

Reply:
Let me know if this is working the way it should now. My understanding of how this function was called was wrong.